packages/location-management/src/components/LocationForm/CustomTreeSelect/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { OpenSRPService } from '@opensrp/react-utils';
+import { loadHierarchy, loadJurisdictions } from '../../../helpers/dataLoaders';
+import { useEffect, useState } from 'react';
+import { baseURL } from '../../../constants';
+import { sendErrorNotification } from '@opensrp/notifications';
+import React from 'react';
+import { TreeNode } from '../../../ducks/locationHierarchy/types';
+import { LocationUnit } from '../../../ducks/location-units';
+import { generateJurisdictionTree } from '../../../ducks/locationHierarchy/utils';
+import { TreeSelect } from 'antd';
+import { TreeSelectProps } from 'antd/lib/tree-select/';
+import { LabelValueType, DataNode } from 'rc-tree-select/lib/interface';
+import { treeToOptions } from '../utils';
+
+/** props for service types select component */
+export interface CustomTreeSelectProps extends TreeSelectProps<LabelValueType> {
+  service: typeof OpenSRPService;
+  baseURL: string;
+  fullDataCallback?: (node?: TreeNode) => void;
+}
+
+const defaultProps = {
+  service: OpenSRPService,
+  baseURL: baseURL,
+};
+
+/** form field where user can select the parent location from a tree structure
+ *
+ * @param props - the component props
+ */
+const CustomTreeSelect = (props: CustomTreeSelectProps) => {
```

Comment:
I presume this is a resuable location tree select that can be used by other packages e.g card-support?

Reply:
I think there was an issue to make an exportable and reusable `TreeSelect` in location management that could be used by [card-support](https://github.com/OpenSRP/web/blob/master/packages/card-support/src/components/DownloadClientData/index.tsx#L171)

Reply:
ok, I will create an issue to move this to react-utils, if that's ok.